ompi/mca/pml/ob1/custommatch/pml_ob1_custom_match_arrays.h

```diff
@@ -0,0 +1,589 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2018      Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2018      Sandia National Laboratories.  All rights reserved.
+ *
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#ifndef PML_OB1_CUSTOM_MATCH_ARRAYS_H
+#define PML_OB1_CUSTOM_MATCH_ARRAYS_H
+
+#include <immintrin.h>
+
+#include "../pml_ob1_recvreq.h"
+#include "../pml_ob1_recvfrag.h"
+
+#define PRQ_SIZE 2
+
+typedef struct custom_match_prq_node
+{
+    int32_t tags[PRQ_SIZE];
```

Comment:
These fields seems to always be accessed together, but laying out the structure this way guarantee 5 cache misses per access which is rather expensive for such a time critical operation.

Reply:
I agree that restructuring would be pertinent to allow for performant PRQ_SIZE and UMQ_SIZE resizing. The current implementation of 'pml_ob1_custom_match_arrays.h' matching engine structure is explicitly sized to fit each prq/umq node into a single cache line.